x-pack/plugins/enterprise_search/public/applications/workplace_search/components/overview/overview_logic.ts

```diff
@@ -170,12 +159,11 @@ export const OverviewLogic = kea({
     hasErrorConnecting: [
       false,
       {
-        setHasErrorConnecting: (_: IOverviewServerData, { hasErrorConnecting }: IOverviewValues) =>
-          hasErrorConnecting,
+        setHasErrorConnecting: (_, { hasErrorConnecting }) => hasErrorConnecting,
       },
     ],
   }),
-  selectors: ({ selectors }: { selectors: IOverviewValues }) => ({
+  selectors: ({ selectors }): TKeaSelectors<IOverviewValues> => ({
     hideOnboarding: [
```

Comment:
You can test the new type checking here by changing this to something like

```suggestion
  selectors: ({ selectors }): TKeaSelectors<IOverviewValues> => ({
    nonexistentValue: [
```
And you'll now get the VSCode error: `Object literal may only specify known properties, and 'nonexistentValue' does not exist in type 'TKeaSelectors<IOverviewValues>'.`